ts/components/ui/LightModal.tsx

```diff
@@ -0,0 +1,73 @@
+/**
+ * Implements a "lite" Modal using Context API
+ */
+
+import { View } from "native-base";
+import React from "react";
+import { StyleSheet } from "react-native";
+
+interface LightModalContextInterface {
+  component: React.ReactNode;
+  showModal: (component: React.ReactNode) => void;
+  hideModal: () => void;
+}
+
+export const LightModalContext = React.createContext<
+  LightModalContextInterface
+>({
+  component: null,
+  showModal: () => undefined,
+  hideModal: () => undefined
+});
+
+type Props = {};
+
+type State = LightModalContextInterface;
+
+const styles = StyleSheet.create({
+  container: {
+    position: "absolute",
+    left: 0,
+    right: 0,
+    bottom: 0,
+    top: 0,
+    zIndex: 1000
+  }
+});
+
+export const LightModalConsumer = LightModalContext.Consumer;
+
+export class LightModalProvider extends React.Component<Props, State> {
+  public showModal = (component: React.ReactNode) => {
+    this.setState({
+      component
+    });
+  };
+
+  public hideModal = () =>
+    this.setState({
+      component: null
+    });
+
+  public state = {
```

Comment:
why make the state public?

Reply:
The typescript definition of React.Component do not specify a visibility modifier for the state so it is public by default and if i change or omit the visibility here Typescript complains.